parsers/flutter.go

```diff
@@ -0,0 +1,67 @@
+package parsers
+
+import (
+	"encoding/json"
+	"github.com/cirruslabs/cirrus-ci-annotations/model"
+	"io"
+	"os"
+	"strings"
+)
+
+type flutterTest struct {
+	Name   string
+	URL    string
+	Line   int64
+	Column int64
+}
+
+type flutterEntry struct {
+	Type string
+	Test flutterTest
+}
+
+func ParseFlutterAnnotations(path string) (error, []model.Annotation) {
+	file, err := os.Open(path)
+	if err != nil {
+		return err, nil
+	}
+
+	decoder := json.NewDecoder(file)
+	result := make([]model.Annotation, 0)
+
+	for {
+		var entry flutterEntry
+
+		if err := decoder.Decode(&entry); err != nil {
+			if err == io.EOF {
+				break
+			}
+
+			return err, nil
+		}
+
+		// Does it look like a TestStartEvent[1] with a valid file associated with it?
+		//
+		// [1]: https://github.com/dart-lang/test/blob/master/pkgs/test/doc/json_reporter.schema.json
+		if entry.Type != "testStart" || entry.Test.URL == "" {
```

Comment:
Should we look save the test information here and only on entries like `"testID":6,"result":"success"` should we create an annotation?

Here it seems creates a warning annotation for a successful test in the future. Check out this failing task for an example of the report containing failures https://cirrus-ci.com/task/5093820242067456

Reply:
Indeed, thanks, see https://github.com/cirruslabs/cirrus-ci-annotations/pull/29/commits/487b5a8485ff450a173b1a646605457b8b39e95a.